src/map/mob.cpp

```diff
@@ -1154,7 +1154,10 @@ int mob_spawn (struct mob_data *md)
 
 	md->state.aggressive = status_has_mode(&md->status,MD_ANGRY)?1:0;
 	md->state.skillstate = MSS_IDLE;
-	md->next_walktime = tick+rnd()%1000+MIN_RANDOMWALKTIME;
+	if (battle_config.mob_walk_onspawn)
+		md->next_walktime = tick+rnd()%1000+MIN_RANDOMWALKTIME;
+	else
+		md->next_walktime = 0;
```

Comment:
```suggestion
		md->next_walktime = -1;
```
Actually you might want to make this to -1, because there is one diff check that looks for <0.